src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java

```diff
@@ -445,14 +457,71 @@ public HttpRequest removeParams()
     }
 
     /**
-     * Sets the body of this request.
+     * Sets the given string as the textual body of this request. Any textual or binary body set previously will be
+     * discarded.
      *
      * @param body
      *            the request body as string
      */
     public HttpRequest body(final String body)
     {
         this.body = body;
+        this.bytesBody = null;
+
+        return this;
+    }
+
+    /**
+     * Sets the given byte array as the binary body of this request. Any textual or binary body set previously will be
+     * discarded.
+     *
+     * @param bytes
+     *            the request body as byte array
+     */
+    public HttpRequest body(final byte[] bytes)
+    {
+        this.bytesBody = bytes;
+        this.body = null;
+
+        return this;
+    }
+
+    /**
+     * Sets the content of the given file as the binary body of this request. Any textual or binary body set previously
+     * will be discarded.
+     * <p>
+     * Note: This method reads the file completely into memory and afterwards calls {@link #body(byte[])} with the data
+     * read.
+     *
+     * @param file
+     *            the file from which to read the request body
+     * @throws IOException
+     *             if the file could not be read
+     */
+    public HttpRequest body(final File file) throws IOException
+    {
+        final byte[] bytes = FileUtils.readFileToByteArray(file);
+        body(bytes);
+
+        return this;
```

Comment:
Can be shortened to `return body(bytes);` but doesn't have to.

Reply:
Done.